chef_master/source/server_security.rst

```diff
@@ -310,6 +310,119 @@ These newer add-ons will also write all of their secrets to ``/etc/opscode/priva
 
+SSL Encryption Between Chef Infra Server and External PostgreSQL
+================================================================
+
+**New in Chef Infra Server 13.1.13:**  Chef Infra Server 13.1.13 introduces the ability to encrypt traffic between Chef Infra Server and an external PostgreSQL server over SSL.  These instructions are not all-encompassing and assume some familiarity with PostgreSQL administration, configuration, and troubleshooting. Consult the `PostgreSQL documentation <https://www.postgresql.org/docs/9.6/ssl-tcp.html>`_ for more information.
+
+Here is a walkthrough of a typical scenario for enabling encryption between a Chef Infra Server machine and an external PostgreSQL machine.  It is assumed that both machines are networked together and user-accessible.
+
+#. Run the following command on both machines to gain root access:
+
+   .. code-block:: bash
+    
+      sudo -i
+
+#. Ensure that `OpenSSL <https://www.openssl.org>`_ is installed on the PostgreSQL machine.
+
+#. Ensure that PostgreSQL is installed on the PostgreSQL machine and has SSL support compiled-in.  This applies whether you are compiling your own source or using a pre-compiled binary.
+
+#. Place SSL certificates in the proper directories on the PostgreSQL machine, and ensure they have correct filenames, ownerships, and permissions.
+
+#. Enable SSL on PostgreSQL, and specify paths to the SSL certificates.  This can be done by editing ``postgresql.conf`` on the PostgreSQL machine and ensuring the relevant entries are present (substitute the appropriate paths):
+
+   .. code-block:: text
+   
+      ssl=on
+   
+      ssl_cert_file='/path/to/cert/file'
+      ssl_key_file='/path/to/key/file'
+
+#. To prevent PostgreSQL from accepting non-SSL connections, edit ``pg_hba.conf`` on the PostgreSQL machine and change the relevant Chef Infra Server connections to `hostssl`.
+
+   Here is a sample ``pg_hba.conf`` file with `hostssl` connections for Chef Infra Server (the contents of your ``pg_hba.conf`` will be different):
+
+   .. code-block:: text
+   
+      # "local" is for Unix domain socket connections only
+      local      all             all                                     peer
+   
+      # IPv4 local connections:
+      hostssl    all             all             127.0.0.1/32            md5
+   
+      # IPv6 local connections:
+      hostssl    all             all             ::1/128                 md5
+   
+      # nonlocal connections
+      hostssl    all             all            192.168.33.100/32        md5
+
+#. Restart PostgreSQL.  This can typically be done with the following command on the PostgreSQL machine (substitute the appropriate path):
+
+   .. code-block:: bash
+   
+      $ /path/to/postgresql/postgresql restart
+
+#. Edit ``/etc/opscode/chef-server.rb`` on the Chef Infra Server machine and add the following line:
```

Comment:
```suggestion
#. Edit ``/etc/opscode/chef-server.rb`` on the Chef Infra Server and add the following line:
```

Reply:
Same 'machine' discussion.  i'm fine committing your edits if you are fine with them.